docs/samples/Microsoft.ML.Samples/Dynamic/LightGBM/LightGBMMulticlassClassification.cs

```diff
@@ -0,0 +1,81 @@
+            // as a catalog of available operations and as the source of randomness.
+            var mlContext = new MLContext();
+
+            // Create in-memory examples as C# native class.
+            var examples = DatasetUtils.GenerateRandomMulticlassClassificationExamples(1000);
+
+            // Convert native C# class to IDataView, a consumble format to ML.NET functions.
+            var dataView = mlContext.Data.ReadFromEnumerable(examples);
+
+            // Create a pipeline. 
+            //  - Convert the string labels into key types.
+            //  - Apply LightGbm multiclass trainer
+            var pipeline = mlContext.Transforms.Conversion.MapValueToKey("LabelIndex", "Label")
+                        .Append(mlContext.MulticlassClassification.Trainers.LightGbm(labelColumn: "LabelIndex"))
+                        .Append(mlContext.Transforms.Conversion.MapValueToKey("PredictedLabelIndex", "PredictedLabel"))
+                        .Append(mlContext.Transforms.CopyColumns("Scores", "Score"));
+
+            // Split the static-typed data into training and test sets. Only training set is used in fitting
+            // the created pipeline. Metrics are computed on the test.
+            var (trainingData, testingData) = mlContext.MulticlassClassification.TrainTestSplit(dataView, testFraction: 0.5);
+
+            // Train the model.
+            var model = pipeline.Fit(trainingData);
+
+            // Do prediction on the test set.
+            var dataWithPredictions = model.Transform(testingData);
+
+            // Evaluate the trained model is the test set.
+            var metrics = mlContext.MulticlassClassification.Evaluate(dataWithPredictions, label: "LabelIndex");
+
+            // Check if metrics are resonable.
+            Console.WriteLine("Macro accuracy: {0}, Micro accuracy: {1}.", 0.863482146891263, 0.86309523809523814);
+
+            // Convert prediction in ML.NET format to native C# class.
+            var nativePredictions = mlContext.CreateEnumerable<DatasetUtils.MulticlassClassificationExample>(dataWithPredictions, false).ToList();
+
+            // Get schema object out of the prediction. It contains metadata such as the mapping from predicted label index
+            // (e.g., 1) to its actual label (e.g., "AA"). The call to "AsDynamic" converts our statically-typed pipeline into
+            // a dynamically-typed one only for extracting metadata. In the future, metadata in statically-typed pipeline should
+            // be accessible without dynamically-typed things.
+            var schema = dataWithPredictions.Schema;
+
+            // Retrieve the mapping from labels to label indexes.
+            var labelBuffer = new VBuffer<ReadOnlyMemory<char>>();
+            schema[nameof(DatasetUtils.MulticlassClassificationExample.PredictedLabelIndex)].Metadata.GetValue("KeyValues", ref labelBuffer);
```

Comment:
>KeyValues [](start = 112, length = 9)

a note about where to find this string.

Reply:
we actually have GetKeyValues on top of metadata, use it!

---
In reply to: [255276416](https://github.com/dotnet/machinelearning/pull/2476#discussion_r255276416) [](ancestors = 255276416)